tools/concat_ice_daily.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+#concatenate sea-ice daily output
+#script inspired from https://github.com/COSIMA/1deg_jra55_ryf/blob/master/sync_data.sh#L87-L108
+
+for d in archive/output*/ice/OUTPUT; do
+    for f in $d/iceh.????-??-01.nc; do
+        if [[ ! -f ${f/-01.nc/-IN-PROGRESS} ]] && [[ ! -f ${f/-01.nc/-daily.nc} ]];
+        then
+            touch ${f/-01.nc/-IN-PROGRESS}
+            echo "doing ncrcat -O -L 5 -4 ${f/-01.nc/-??.nc} ${f/-01.nc/-daily.nc}"
+            ${PAYU_PATH}/ncrcat -O -L 5 -4 ${f/-01.nc/-??.nc} ${f/-01.nc/-daily.nc} && chmod g+r ${f/-01.nc/-daily.nc} && rm ${f/-01.nc/-IN-PROGRESS}
+            if [[ ! -f ${f/-01.nc/-IN-PROGRESS} ]] && [[ -f ${f/-01.nc/-daily.nc} ]];
+            then
+                for daily in ${f/-01.nc/-??.nc}
+                do
+                    # mv $daily $daily-DELETE  # rename individual daily files - user to delete
+                    rm $daily
+                done
+            else
+                rm ${f/-01.nc/-IN-PROGRESS}
+            fi
+        fi
+    done
```

Comment:
```suggestion
    # Don't error if there are no matching patterns
    shopt -s nullglob
    
    # Assuming `$d` contains the directory where the data resides
    
    for first_file in $d/iceh.????-??-01.nc
    do
      # Make a list of all files we wish to concatenate
      icefiles=(${first_file/-01.nc/-??.nc})
    
      if [ ${#icefiles[@]} -gt 0 ]
      then
        iceout="${first_file/-*.nc/-daily.nc}"
        ncrcat -O -L 5 -4 "${icefiles[@]}" ${iceout} && rm "${icefiles[@]}"
      fi
    done
```

Personally I prefer to just delete the files if the return status of the `ncrcat` command is ok. Making temporary files ends up introducing extra logic to deal with them.

Note the above is untested, just a suggestion for how to reduce the complexity of the logic.

Reply:
> Making temporary files ends up introducing extra logic to deal with them.

This is copied from the COSIMA scripts. I assume the temporary files were needed for some edge case? @aekiss - Do you know why the temporary files were used?

Reply:
Sorry to badger you @aekiss but I'm also curious if there were cases of data loss that prompted the design you implemented?

Reply:
Andrew said it was just for sanity / checking debug in case of failure. I am happy to remove it.